sdk/communication/azure-communication-common-perf/src/main/java/com.azure.communication.common.perf/HmacAuthenticationPolicyTest.java

```diff
@@ -0,0 +1,49 @@
+package com.azure.communication.common.perf;
+
+import com.azure.communication.common.implementation.HmacAuthenticationPolicy;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.http.*;
+import com.azure.perf.test.core.PerfStressOptions;
+import com.azure.perf.test.core.PerfStressTest;
+import reactor.core.publisher.Mono;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class HmacAuthenticationPolicyTest extends PerfStressTest<PerfStressOptions> {
+
+    private final HmacAuthenticationPolicy hmacAuthenticationPolicy;
+    private final HttpPipeline pipeline;
+    private final HttpRequest request;
+
+    class NoOpHttpClient implements HttpClient {
+        @Override
+        public Mono<HttpResponse> send(HttpRequest request) {
+            return Mono.empty(); // NOP
+        }
+    }
+
+    public HmacAuthenticationPolicyTest(PerfStressOptions options) throws MalformedURLException {
+        super(options);
+        String mockedKey = "JdppJP5eH1w/CQ0cx4RGYWoC7NmQ0nmDbYR2PYWSDTXojV9bI1ck0Eh0sUIg8xj4KYj7tv+ZPLICu3BgLt6mMz==";
+        AzureKeyCredential keyCredential = new AzureKeyCredential(mockedKey);
+        hmacAuthenticationPolicy = new HmacAuthenticationPolicy(keyCredential);
+        pipeline = new HttpPipelineBuilder()
+            .httpClient(new NoOpHttpClient())
+            .policies(hmacAuthenticationPolicy)
+            .build();
+
+        request = new HttpRequest(HttpMethod.GET, new URL("https://test.com/"));
```

Comment:
The original problem was the state variable `sha256HMAC`. Due to a race condition the signature could have been calculated incorrectly. 
So we need to start sending different payload so that the `stringToSign` in HmacAuthenticationPolicy.java is always different and verify that the stress test can discover the fact that it has indeed been calculated incorrectly (so we'll need to be able to calculate correct values as well somewhere to be able to tell the difference).